crates/core/src/dht/dht_store.rs

```diff
@@ -306,44 +306,30 @@ impl DhtStore {
     }
 }
 
-use petgraph::{graph::DiGraph, prelude::NodeIndex, Direction::Outgoing};
-use std::collections::HashMap;
+use std::collections::HashSet;
```

Comment:
Could we use immutable crate here? it is much faster imo

Reply:
[citation needed] ;)

Reply:
https://docs.rs/im/14.0.0/im/

"The maps - HashMap and OrdMap - generally perform similarly to their equivalents in the standard library, but tend to run a bit slower on the basic operations (HashMap is almost neck and neck with its counterpart, while OrdMap currently tends to run 2-3x slower). On the other hand, they offer the cheap copy and structural sharing between copies that you'd expect from immutable data structures."

-having the hashmap be almost neck and neck but benefiting from structural sharing if we ever want to do a clone makes it look more efficient from my perspective but I could be wrong.

We could also just say no more cloning but use of the std hashmap will be slower if a clone is ever introduced